clients/src/main/java/org/apache/kafka/common/requests/ShareFetchRequest.java

```diff
@@ -65,8 +65,11 @@ public static Builder forConsumer(int maxWait, int minBytes, Map<TopicPartition,
                     data.topics().add(fetchTopic);
                 }
 
+                // TODO: Remove the hardcoded setPartitionMaxBytes once this field is populated correctly by the client.
+                //  It was added to confirm that the integration tests pass and we are able to fetch records
                 ShareFetchRequestData.FetchPartition fetchPartition = new ShareFetchRequestData.FetchPartition()
                         .setPartitionIndex(topicPartition.partition())
+                        .setPartitionMaxBytes(40000)
```

Comment:
You can merge with this hard-coded value. Open a JIRA in the current sprint against me to put proper support into the client.